src/fenced/fenced/logging.py

```diff
@@ -26,6 +28,19 @@ def handleError(self, record):
             self.sock = None
 
 
+def ensure_logdir_exists():
+    """
+    We need to ensure that the directory in `LOG_FILE` exists
+    so logging works
+    """
+    dirname = os.path.dirname(LOG_FILE)
+
+    if not os.path.exists(dirname):
+        os.makedirs(dirname, exist_ok=True)
+
+    return
+
+
 def setup_logging(foreground):
```

Comment:
And maybe just call ensure_logdir_exists in here? Then we dont have to change main.py